src/data/pnat/basic.lean

```diff
@@ -349,4 +350,122 @@ theorem gcd_mul_lcm (n m : ℕ+) : (gcd n m) * (lcm n m) = n * m :=
 
 def prime (p : ℕ+) : Prop := (p : ℕ).prime
 
+/-- Two pnats are coprime if their gcd is 1. -/
+def coprime (m n : ℕ+) : Prop := m.gcd n = 1
+
+@[simp]
+lemma eq_iff {m n : ℕ+} :
```

Comment:
This should be called `coe_inj`, and it might already be available as `ext_iff` but with the two sides swapped. Also, it should move up to somewhere just after the definition of the coercion.

Reply:
I found it under set_coe.ext_iff. I did use it once as a simp lemma, so I'll leave it in with the simpler proof, as I think it's nice to have simp reduce that, unless you think it's better to just simp [set_coe.ext_iff].

Reply:
It's fine to keep this statement, especially because you want to tag it with `@[simp]`. But the name should be `coe_inj`. Because that is what's used for similar lemmas in the rest of the library, and hence predicatable.
(I think that `pnat.eq` is a bad guy, it should be renamed to `pnat.ext` to follow naming conventions.)
Also, you should be able to prove this in half a line with this new lemma that you found.